src/interval.jl

```diff
@@ -279,22 +279,24 @@ end
 
 # There is power in a union.
 """
-    union(intervals::AbstractVector{<:AbstractInterval})
+    union(intervals::Union{AbstractInterval, AbstractVector{<:AbstractInterval}}, itrs...)
 
-Flattens a vector of overlapping intervals into a new, smaller vector containing only
+Flattens sets of overlapping intervals into a new, smaller vector containing only
 non-overlapping intervals.
 """
-function Base.union(intervals::AbstractVector{<:AbstractInterval})
-    return union!(convert(Vector{AbstractInterval}, intervals))
+function Base.union(
+    interval::Union{AbstractInterval, AbstractVector{<:AbstractInterval}}, intervals...,
+)
+    return union!(AbstractInterval[interval; intervals...])
 end
```

Comment:
Seems odd we would support `union([Interval(1,2)], Interval(2,3))`

Reply:
The problem with Intervals is that an Interval is just as much a set as a vector of intervals. It doesn't make sense to me to only allow vectors.

Reply:
I was probably not very clear. I find it weird we are supporting mixing vectors of intervals and intervals. My preference would be to have:
```julia
Base.union(intervals::AbstractVector{<:AbstractInterval}) = ...
Base.union(intervals::AbstractInterval...) = ...
```

Reply:
My point still stands.

Also, vcat has the same behaviour.

Reply:
I'm not a fan of this behaviour but we should stick to how `union` currently behaves in `Base.